File: 7-reverse-integer/reverse-integer.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(n) {
     let revNum=0;
        while(n!=0){
            let digit=n%10;
            revNum=revNum*10+digit;
            n=Math.trunc(n/10);
        }
        if(revNum<-(2**31)||revNum>(2**31)-1) return 0;
        return revNum;
};

File: 7-reverse-integer/reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        sign= -1 if x<0 else 1;
        num=abs(x);
        if num<10:
            return x
        revNum=0;
        while num>0:
            ld=num%10;
            revNum=revNum*10 +ld;
            num=num//10;
        revNum*=sign;
        if revNum<-(2**31) or revNum>(2**31-1):
            return 0;
        return revNum;
